### DISCUSSION TITLE
/posts/2023/04/17%E6%88%91%E7%BB%88%E4%BA%8E%E4%BC%9A%E7%94%A8Docker%E4%BA%86(nest+prisma+psotgresql+nginx+https)

### DISCUSSION THREAD

**Discussion by giscus** [Justin3go/justin3go.com#44] [Announcements]
# /posts/2023/04/17%E6%88%91%E7%BB%88%E4%BA%8E%E4%BC%9A%E7%94%A8Docker%E4%BA%86(nest+prisma+psotgresql+nginx+https)

坚持深耕技术领域的T型前端程序员, 关注独立开发与AI，喜欢Vuejs、Nestjs, 还会点Python、搜索引擎、NLP、Web3、后端

https://justin3go.com/posts/2023/04/17%E6%88%91%E7%BB%88%E4%BA%8E%E4%BC%9A%E7%94%A8Docker%E4%BA%86(nest+prisma+psotgresql+nginx+https)

**Comment by nonhana** [comment]
emmm，您好，请问看得见这条评论吗~~？
我按照您的方式，在配置 https 的时候也是借鉴的您的这篇文章，但是我把我映射到本地的端口号给更改了一下。因为原来的443和80被我用来部署个人博客。
我分别将这两个端口映射为本地的721和8000。
然后进行 docker-compose up，能够成功的将服务跑起来。

但是问题是，为什么当我使用域名访问时，却显示连接错误呢？使用ip地址访问的时候是一切正常的。

![报错](https://common-1319721118.cos.ap-shanghai.myqcloud.com/picgo/20240601182326.png)

是我的 nginx 配置不对还是其他什么原因呢？请问您自己是哪个端口进行访问的呢？

**Reply by Justin3go** [reply]
可以检查一下nginx是否有相关的错误日志吗？在博客中配置的log路径里面可以看到。

**Reply by nonhana** [reply]
您好，我检查了一下 nginx 容器内部，发现原来它是通过 /etc/conf.d/default.conf 来跑的，而不是我服务器上面 nginx 的配置 /www/server/nginx/conf/nginx.conf 来跑的，我在配置的时候误以为两者是属于同一个东西，因此没有把 conf.d 复制过去。
现在我已经成功的配置 default.conf 文件，将服务跑起来了！感谢您的回复哈哈，您的文章让我受益良多！

**Comment by Hbisedm** [comment]
您好， 我数据库是mysql，容器内的nest的prisma一直拿不到mysql的服务，好奇怪，redis却可以，不知道是不是prisma的问题

**Reply by Hbisedm** [reply]
docker-compose 里面的mysql配置得加  
    restart: always
就解决了
